sqlglot/dialects/snowflake.py

```diff
@@ -143,6 +143,14 @@ def _datatype_sql(self: generator.Generator, expression: exp.DataType) -> str:
     return self.datatype_sql(expression)
 
 
+def _parse_convert_timezone(args: t.Sequence) -> exp.Expression:
+    if len(args) == 3:
+        converted: t.List[exp.Expression] = [maybe_parse(arg, dialect=Snowflake) for arg in args]
```

Comment:
Is there a reason you used `maybe_parse` here? The `args` parameter should aready be the parsed args.